base_location/models/res_company.py

```diff
@@ -57,12 +57,9 @@ def _onchange_zip_id(self):
         if self.zip_id:
             self.zip = self.zip_id.name
             self.city_id = self.zip_id.city_id
-            self.city = self.zip_id.city
-            self.country_id = self.zip_id.country_id
-            if self.country_id.enforce_cities:
-                self.state_id = self.city_id.state_id
-            else:
-                self.state_id = self.zip_id.state_id
+            self.city = self.zip_id.city_id.name
```

Comment:
On v11, I remember this to be problematic, not filling correctly fields, but as it's only one time operation, I ignore the problem. Have you checked if this works?

Reply:
Yup, it works. Tried several times and it worked for me. Will try more times in case it fails.